### PR TITLE
No eval for OpenCL - One last thing on my ethminer todo list.

### PR DESCRIPTION
**Pull request by jean-m-cyr**
- Rename -cuda-noeval option to --noeval since it can now apply to
  both Cuda and OpenCl.
- Add support for using GPU mix hash result when noeval is set in
  CLMiner.
- Modify experimental opencl kernel to return mix hash on solution
  found.
- Modify stable opencl kernel to return mix hash on solution
  found.
- Move global s_noval bool to common miner.cpp

### CODE REVIEW

**Comment by jean-m-cyr** [comment]
Appveyor build appears broken!

File: libethash-cl/CLMiner.cpp [hunk 1]
```diff
@@ -359,14 +363,21 @@ void CLMiner::workLoop()
...
-                    farm.submitProof(Solution{nonce, r.mixHash, current, current.header != w.header});
+            if (results.count) {
+                uint64_t nonce = current.startNonce + results.gid;
+                if (!s_noeval) {
+                    Result r = EthashAux::eval(current.epoch, current.header, nonce);
+                    if (r.value < current.boundary)
+                        farm.submitProof(Solution{nonce, r.mixHash, current, current.header != w.header});
+                    else {
+                        farm.failedSolution();
+                        cwarn << "discarded incorrect result!";
```
Comment (smurfy) [inline libethash-cl/CLMiner.cpp]: Please use same message as CUDA
``cwarn << "GPU gave incorrect result!";``

**Comment by chfast** [comment]
AppVeyor build fixed in https://github.com/ethereum-mining/ethminer/pull/1047.

**Comment by chfast** [comment]
Can you rebase to clean the windows build?

**Comment by jean-m-cyr** [comment]
Sure

**Comment by jean-m-cyr** [comment]
Sure. I'll let the current build finish to verify chfast's build fix, then do it.

**Comment by jean-m-cyr** [comment]
@AndreaLanfranchi done. --cuda-noeval restored for cuda builds.

**Comment by AndreaLanfranchi** [comment]
Great.

**Pull request merged**